--- a/unittest/my_decimal/Makefile.am	2013-12-12 09:37:48.508314567 -0500
+++ b/unittest/my_decimal/Makefile.am	2013-12-12 09:38:28.187519854 -0500
@@ -18,11 +18,11 @@
 AM_CPPFLAGS     += -I$(top_srcdir)/sql
 
 LDADD 		= $(top_builddir)/unittest/mytap/libmytap.a \
-		  $(top_builddir)/mysys/libmysys.a \
-		  $(top_builddir)/strings/libmystrings.a \
-		  $(top_builddir)/dbug/libdbug.a \
-		  $(top_builddir)/mysys/libmysys.a \
-		  $(top_builddir)/strings/libmystrings.a
+		  $(top_builddir)/mysys/libmysys.la \
+		  $(top_builddir)/strings/libmystrings.la \
+		  $(top_builddir)/dbug/libdbug.la \
+		  $(top_builddir)/mysys/libmysys.la \
+		  $(top_builddir)/strings/libmystrings.la
 
 my_decimal_t_SOURCES = my_decimal-t.cc
 
