diff -aurN a/cmake/maintainer.cmake b/cmake/maintainer.cmake
--- a/cmake/maintainer.cmake	2014-11-21 00:39:51.000000000 -0500
+++ b/cmake/maintainer.cmake	2014-12-03 13:19:50.893380789 -0500
@@ -34,13 +34,6 @@
       "${MY_CXX_WARNING_FLAGS} -Wno-null-conversion -Wno-unused-private-field")
 ENDIF()
 
-# Turn on Werror (warning => error) when using maintainer mode.
-IF(MYSQL_MAINTAINER_MODE)
-  SET(MY_C_WARNING_FLAGS "${MY_C_WARNING_FLAGS} -Werror")
-  SET(MY_CXX_WARNING_FLAGS "${MY_CXX_WARNING_FLAGS} -Werror")
-  SET(COMPILE_FLAG_WERROR 1)
-ENDIF()
-
 # Set warning flags for GCC/Clang
 IF(CMAKE_COMPILER_IS_GNUCC OR CMAKE_C_COMPILER_ID MATCHES "Clang")
   SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MY_C_WARNING_FLAGS}")
