--- CMakeLists.orig.txt 2013-01-18 16:58:34.701624830 -0500
+++ CMakeLists.txt      2013-01-18 17:00:51.479636903 -0500
@@ -294,7 +294,11 @@
 ADD_SUBDIRECTORY(regex)
 ADD_SUBDIRECTORY(mysys)
 ADD_SUBDIRECTORY(libmysql)
-
+ADD_SUBDIRECTORY(client)
+ADD_SUBDIRECTORY(libservices)
+ADD_SUBDIRECTORY(scripts)
+ADD_SUBDIRECTORY(sql/share)
+ADD_SUBDIRECTORY(support-files)
 
 IF(WITH_UNIT_TESTS)
  ENABLE_TESTING()
@@ -307,10 +311,7 @@
 ADD_SUBDIRECTORY(extra)
 IF(NOT WITHOUT_SERVER)
   ADD_SUBDIRECTORY(tests)
-  ADD_SUBDIRECTORY(client)
   ADD_SUBDIRECTORY(sql)
-  ADD_SUBDIRECTORY(sql/share)
-  ADD_SUBDIRECTORY(libservices)
   OPTION (WITH_EMBEDDED_SERVER "Compile MySQL with embedded server" OFF)
   IF(WITH_EMBEDDED_SERVER) 
    ADD_SUBDIRECTORY(libmysqld)
@@ -319,19 +320,18 @@
 
   ADD_SUBDIRECTORY(mysql-test)
   ADD_SUBDIRECTORY(mysql-test/lib/My/SafeProcess)
-  ADD_SUBDIRECTORY(support-files)
-  ADD_SUBDIRECTORY(scripts)
   ADD_SUBDIRECTORY(sql-bench)
-  IF(UNIX)
-    ADD_SUBDIRECTORY(man)
-  ENDIF()
   IF(EXISTS ${CMAKE_SOURCE_DIR}/internal/CMakeLists.txt)
     ADD_SUBDIRECTORY(internal)
   ENDIF()
   ADD_SUBDIRECTORY(packaging/rpm-uln)
   ADD_SUBDIRECTORY(packaging/rpm-oel)
 ENDIF()
 
+IF(UNIX)
+  ADD_SUBDIRECTORY(man)
+ENDIF()
+
 INCLUDE(cmake/abi_check.cmake)
 INCLUDE(cmake/tags.cmake)

